src/BoilerPlate/App.cpp

```diff
@@ -206,6 +217,67 @@ namespace Engine
 		}
 	}
 
+	unsigned int power_two_floor(unsigned int val) {
+		unsigned int power = 2, nextVal = power * 2;
+		while ((nextVal *= 2) <= val)
+			power *= 2;
+		return power * 2;
+	}
+
+	
+
+	void RenderText(std::string message, SDL_Color color, float x, float y)
```

Comment:
This is minimal code for text rendering, please clean it up. Use your own Color class and such.